### ISSUE TITLE
Flipper Zero App resets favorites when synced

### ISSUE THREAD

**Issue by alex1001458** [flipperdevices/flipperzero-firmware#2174]
### Describe the bug.

Every application or file in your favorites tab the flipper zero app doesn't recognize in it's archive menu (eg. any app inside a folder) will get removed from the favorites tab when you sync it with the flipper zero.

### Reproduction

1. "Pin" something through the quick menu (press down on screen saver) the Flipper app doesn't recognize, such as an IR remote you have in a folder.
2. Turn on Flipper app and let it synchronize (you may have to press "synchronize" yourself)
3. Watch your favorited item disappear.

**Comment by skotopes** [comment]
Are we talking about mobile apps?

**Comment by alex1001458** [comment]
> Are we talking about mobile apps?

No, sorry for lack of clarification, I'm talking about any program you have on your flipper zero

**Comment by skotopes** [comment]
Then what do you mean by "synchronize"?

**Comment by skotopes** [comment]
@alex1001458 still need more details from you ;-) Please provide details on how to reproduce issue with exact firmware/platform/app versions been used.

**Comment by alex1001458** [comment]
Sorry for lack of response. I'm just going to use pictures to simplify.
1.My Favorites before opening the phone app
![image](https://user-images.githubusercontent.com/66147299/209870903-17f38ce1-619f-4b5d-8b72-51a32d6c173d.png)

2. When opening the phone app it only recognizes one of them.
![Screenshot_20221228_155513_Flipper](https://user-images.githubusercontent.com/66147299/209871313-5a4c7199-6c99-4fb8-a1db-73486221ddbc.png)

3. The flipper Syncs to what the phone recognizes, removing the other 2 apps.
![image](https://user-images.githubusercontent.com/66147299/209871448-d7594c1a-3ac6-4a73-b229-046b3b97dbef.png)

I suspect it's due to the fact the other 2 favorites are within folders
For example:
![Screenshot 2022-12-28 160206](https://user-images.githubusercontent.com/66147299/209871645-e085317d-94e0-47b6-aea8-79a2ec90ae3c.png)

**Comment by skotopes** [comment]
We talking about Android? Which version of firmware and companion app you are using?

**Comment by alex1001458** [comment]
I am on Android, latest version of the companion app and I am using the unlshd-022 firmware.

**Comment by skotopes** [comment]
Do you have this issue with original firmware?

**Comment by alex1001458** [comment]
Yes.

![image](https://user-images.githubusercontent.com/66147299/209894833-7252f444-4097-4c41-862d-61f2e9356373.png)
![image](https://user-images.githubusercontent.com/66147299/209894936-2d3e0355-432a-4d0e-90c6-08bd98f872fe.png)
